coins/monero/tests/scan.rs

```diff
@@ -0,0 +1,160 @@
+use std::collections::HashSet;
+use hex_literal::hex;
+
+use curve25519_dalek::{constants::ED25519_BASEPOINT_TABLE, scalar::Scalar};
+use rand_core::OsRng;
+use zeroize::Zeroizing;
+
+use monero_serai::{
+  wallet::{
+    address::{Network, MoneroAddress},
+    ViewPair, SignableTransaction, Scanner,
+  },
+};
+
+mod runner;
+
+const ADDR_SPEND: [u8; 32] =
+  hex!("bf02a79d9e30ea76872e565722517924151793c535832e4353e54dc0be698001");
+const ADDR_VIEW: [u8; 32] =
+  hex!("c300b798c000f3c5da4ebc5a288418fe9060d623c9830b7abc3d9e7a3152eb08");
+
+pub struct AddressInfo {
+  address: String,
+  payment_id: Option<[u8; 8]>,
+}
+
+pub async fn scan_incomings(addr_scanner: &mut Scanner, addresses: &Vec<AddressInfo>) {
```

Comment:
While I'm not against this custom runner, shouldn't we define this as a macro around the existing macro, and just run it on one address type at a time?